src/python/pants/backend/jvm/tasks/analysis_extraction.py

```diff
@@ -47,106 +46,89 @@ def _create_products_if_should_run(self):
-        if not vt.valid:
-          self._extract_analysis(vt.target, analysis_file, summary_json_file)
-        self._register_products(vt.target,
-                                cp_entry,
-                                summary_json_file,
-                                classes_by_source,
-                                product_deps_by_src)
-
-  def _extract_analysis(self, target, analysis_file, summary_json_file):
-    target_classpath = self._zinc.compile_classpath('runtime_classpath', target)
-    analysis_by_cp_entry = self._analysis_by_runtime_entry
-    upstream_analysis = list(self._upstream_analysis(target_classpath, analysis_by_cp_entry))
-    args = [
-        '-summary-json', summary_json_file,
-        '-analysis-cache', analysis_file,
-        '-classpath', ':'.join(target_classpath),
-        '-analysis-map', ','.join('{}:{}'.format(k, v) for k, v in upstream_analysis),
-      ]
-
-    result = self.runjava(classpath=self._zinc.extractor,
-                          main=Zinc.ZINC_EXTRACT_MAIN,
-                          args=args,
-                          workunit_name=Zinc.ZINC_EXTRACTOR_TOOL_NAME,
-                          workunit_labels=[WorkUnitLabel.MULTITOOL])
-    if result != 0:
-      raise TaskError('Failed to parse analysis for {}'.format(target.address.spec),
-                      exit_code=result)
-
-  def _upstream_analysis(self, target_classpath, analysis_by_cp_entry):
-    for entry in target_classpath:
-      analysis_file = analysis_by_cp_entry.get(entry)
-      if analysis_file is not None:
-        yield entry, analysis_file
+    targets = self.context.targets()
+    # complete classpath list
+    deps_classpaths = [
+      cp[1] for target in targets for cp in classpath_product.get_for_target(target) if cp[1]
+    ]
+    with self.aliased_classpaths(deps_classpaths) as deps_classpath_by_alias:
+      with self.invalidated(targets,
+                            fingerprint_strategy=fingerprint_strategy,
+                            invalidate_dependents=True) as invalidation_check:
+        for vt in invalidation_check.all_vts:
+          # class paths for the target we are computing deps for
+          target_cps = [cp[1] for cp in classpath_product.get_for_target(vt.target)]
+
+          jdeps_output_json = self._jdeps_output_json(vt)
+          if not vt.valid:
+            self._run_jdeps_analysis(vt.target, target_cps, deps_classpath_by_alias, jdeps_output_json)
+          self._register_products(vt.target,
+                                  jdeps_output_json,
+                                  product_deps_by_target)
+
+  @memoized_property
+  def _jdeps_summary_line_regex(self):
+    return re.compile(r"^\S+\s->\s(\S+)$")
+
+  def _run_jdeps_analysis(self, target, target_cps, deps_classpath_by_alias, jdeps_output_json):
+    with open(jdeps_output_json, 'w') as f:
+      if target_cps:
+        # TODO should we find an abs path to jdeps in a better way? a jdk path?
```

Comment:
Yea, we should. Remind me to give you a link for that tomorrow.

Reply:
So, jvmdoc gen locates and uses the `tools.jar`: https://github.com/pantsbuild/pants/blob/570fab8d3896619b3005c542ee7e61afbd412e67/src/python/pants/backend/jvm/tasks/javadoc_gen.py#L40-L50 : the only additional thing you need to know is the "main class" of the tool itself... if googling doesn't turn that up, we can pair on finding it.